experimenter/tests/integration/nimbus/pages/demo_app/frontend.py

```diff
@@ -0,0 +1,47 @@
+import time
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from nimbus.pages.base import Base
+
+
+class DemoAppPage(Base):
+    def __init__(self, driver):
+        self.driver = driver
+
+    def navigate_to(self):
+        demo_app_url = "http://demo-app-frontend:3000/"
+        self.driver.get(demo_app_url)
+
+    def wait_for_result_text(self, text_list):
+        xpath_conditions = " | ".join([f".='{text}'" for text in text_list])
+        xpath = f"//h1[{' or '.join(xpath_conditions.split(' | '))}]"
+
+        return WebDriverWait(self.driver, 10).until(
+            EC.presence_of_element_located((By.XPATH, xpath))
+        )
```

Comment:
We have a few built in locator methods located in the Base class here. Could you use those? You can use this instead of having to define your own wait class, unless you use something else besides ```presence_of_element_located```.

```suggestion
    def wait_for_result_text(self, text_list):
        xpath_conditions = " | ".join([f".='{text}'" for text in text_list])
        xpath = f"//h1[{' or '.join(xpath_conditions.split(' | '))}]"

        return self.wait_for_locator(self, xpath, description=None):
```

Reply:
good point